imap_processing/ialirt/process_ephemeris.py

```diff
@@ -0,0 +1,199 @@
+"""
+Find azimuth (degrees), elevation (degrees), and doppler shift (Hz).
+
+Based on ephemeris data and ground station location (longitude, latitude, altitude).
+
+Reference: https://spiceypy.readthedocs.io/en/main/documentation.html.
+"""
+
+import logging
+import typing
+
+import numpy as np
+import spiceypy as spice
+from numpy import ndarray
+
+from imap_processing.spice.kernels import ensure_spice
+from imap_processing.spice.time import et_to_utc, str_to_et
+
+# Logger setup
+logger = logging.getLogger(__name__)
+
+
+def calculate_doppler() -> int:
+    """
+    Calculate the doppler shift. Placeholder for now.
+
+    Returns
+    -------
+    doppler : int
+        Doppler shift. Currently a throwaway value.
+    """
+    return 1
+
+
+@typing.no_type_check
+@ensure_spice
+def latitude_longitude_to_ecef(
+    longitude: float, latitude: float, altitude: float
+) -> ndarray:
+    """
+    Convert geodetic coordinates to rectangular coordinates.
+
+    Earth-Centered, Earth-Fixed (ECEF) coordinates are a Cartesian coordinate system
+    with an origin at the center of the Earth.
+
+    Parameters
+    ----------
+    longitude : float
+        Longitude in decimal degrees. Positive east of prime meridian, negative to west.
+    latitude : float
+        Latitude in decimal degrees. Positive north of equator, negative to south.
+    altitude : float
+        Altitude in kilometers.
+
+    Returns
+    -------
+    rect_coords : ndarray
+        Rectangular coordinates in kilometers.
+    """
+    latitude_radians = np.deg2rad(latitude)
+    longitude_radians = np.deg2rad(longitude)
+
+    # Retrieve Earth's radii from SPICE
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.
+    # (url cont.) bodvrd
+    radii = spice.bodvrd("EARTH", "RADII", 3)[1]
+    equatorial_radius = radii[0]  # Equatorial radius in km
+    polar_radius = radii[2]  # Polar radius in km
+    flattening = (equatorial_radius - polar_radius) / equatorial_radius
+
+    # Convert geodetic coordinates to rectangular coordinates
+    # https://spiceypy.readthedocs.io/en/main/documentation.html#spiceypy.spiceypy.
+    # (url cont.) georec
+    rect_coords = spice.georec(
+        longitude_radians, latitude_radians, altitude, equatorial_radius, flattening
+    )
+
+    return rect_coords
+
+
+@typing.no_type_check
+@ensure_spice
+def calculate_azimuth_and_elevation(
+    longitude: float,
+    latitude: float,
+    altitude: float,
+    observation_time: float,
+    target: str = "IMAP",
```

Comment:
There is a SpiceBody class in `imap_processing.spice.geometry.py` that should be used here.
```suggestion
    target: SpiceBody = SpiceBody.IMAP,
```

Reply:
This switch doesn't make sense to me, since SpiceBody.IMAP returns an integer and the `azlcpo` function expects a string. Maybe I'm misunderstanding the SpiceBody Enum? @subagonsouth